Allow testing action with unlocked fields

DBnR1 opened the pull request:
Enhancement #13071

This allows testing of actions with unlocked fields

DBnR1 committed:
Add unlocked fields to IntegrationTestTrait to allow testing actions with unlocked fields

dereuromark commented:
Looks like you want to rebase this on 3.next

dereuromark commented:
@markstory Or could we also put this into master? This seems to only affect tests part and seems quite BC.

DBnR1 committed:
Add missing doc comments

dereuromark commented on src/TestSuite/IntegrationTestTrait.php:
```diff
@@ -282,6 +282,9 @@ public function enableSecurityToken($unlockedFields = [])
     /**
      * Set list of fields that are excluded from field validation.
+     *
+     * @param array $unlockedFields List of fields that are excluded from field validation.
```
Isnt this `string[]` ?

DBnR1 replied:
I copied the comment from SecureFieldTokenTrait but you are right. Maybe fix it in both files.

dereuromark replied:
I will take care of it in 4.x: https://github.com/cakephp/cakephp/pull/13485
but you are welcome to fix in 3.x

DBnR1 committed:
Fix param doc comments

markstory commented:
@dereuromark Sure. Would you be able to rebase your changes onto `master`?

saeideng commented:
please rebase your branch

saeideng commented on src/TestSuite/IntegrationTestTrait.php:
```diff
@@ -263,13 +270,27 @@ public function configApplication($class, $constructorArgs)
      * @return void
      */
-    public function enableSecurityToken()
+    public function enableSecurityToken($unlockedFields = [])
```
why we need this argument when we can use `$this->setUnlockedFields($unlockedFields);` ??

markstory replied:
Agreed. Having both seems redundant.

DBnR1 replied:
A good point. I can't think of a use case where `$this->setUnlockedFields($unlockedFields);` wouldn't do.

saeideng commented:
I rebased changes onto` master`

ADmad commented on src/TestSuite/IntegrationTestTrait.php:
```diff
@@ -263,13 +270,27 @@ public function configApplication($class, $constructorArgs)
+     * @param string[] $unlockedFields List of fields that are excluded from field validation.
+     * @return void
+     */
+    public function setUnlockedFields($unlockedFields = [])
```
Not a fan of public API changes in bugfix release even if it only affects testsuite. Let's at least add a `@since` tag to the docblock.

ADmad replied:
The argument should be typehinted as `array`.

DBnR1 committed:
Removed parameter and added type hint

Removed param from enableSecurityToken() and added type hint for $unlockedFields parameter in setUnlockedFields method

DBnR1 committed:
Fix tests
